python/tskit/trees.py

```diff
@@ -5106,42 +5106,84 @@ def trim(self, record_provenance=True):
         tables.trim(record_provenance)
         return tables.tree_sequence()
 
-    def subset(self, nodes, record_provenance=True):
+    def subset(
+        self,
+        nodes,
+        record_provenance=True,
+        unchanged_populations=False,
+        keep_unused_populations=False,
```

Comment:
It would be good to mirror the options that we have in ``simplify``, which are ``filter_populations=True``, ``filter_individuals=True``, etc.

Reply:
Unfortunately, we can't exactly mirror `simplify`. Here we've got three options:

1. remove unused pops
2. reorder pops, with unused ones at the end
3. keep pops the same

This is because for `canonicalize` we need to reorder things, but not discard information - that's option (2). Similarly, the option for `individuals` toggles between options (1) and (2). But the `filter_X` flags for `simplify` toggle between options (1) and (3).

Maybe the best thing to do is to mirror `simplify` at the python level, but then leave an extra flag that does option (2) (e.g., `TSK_SUBSET_CANONICALIZE`) at the C level, perhaps not even provided in python?

Reply:
I've gone ahead and done this change.